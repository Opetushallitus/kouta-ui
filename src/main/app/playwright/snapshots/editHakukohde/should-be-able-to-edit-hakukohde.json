{
  "oid": "1.2.3.4.5.6",
  "toteutusOid": "2.1.1.1.1.1",
  "hakuOid": "4.1.1.1.1.1",
  "tila": "tallennettu",
  "nimi": {
    "fi": "Hakukohteen nimi"
  },
  "alkamiskausiKoodiUri": "kausi_k#1",
  "alkamisvuosi": "2024",
  "hakulomake": {},
  "pohjakoulutusvaatimusKoodiUrit": [
    "pohjakoulutusvaatimuskouta_pk#1"
  ],
  "pohjakoulutusvaatimusTarkenne": {
    "fi": "<p>Fi tarkenne</p>"
  },
  "muuPohjakoulutusvaatimus": {},
  "toinenAsteOnkoKaksoistutkinto": true,
  "kaytetaanHaunAikataulua": false,
  "liitteetOnkoSamaToimitusaika": false,
  "liitteetOnkoSamaToimitusosoite": false,
  "liitteidenToimitusosoite": null,
  "liitteidenToimitustapa": null,
  "liitteet": [
    {
      "toimitustapa": "osoite",
      "tyyppiKoodiUri": "liitetyypitamm_4#1",
      "nimi": {
        "fi": "Nimi"
      },
      "toimitusaika": "2011-11-11T10:30",
      "toimitusosoite": {
        "osoite": {
          "osoite": {
            "fi": "Osoite"
          },
          "postinumeroKoodiUri": "posti_00350#2"
        },
        "sahkoposti": "sahkoposti@email.com",
        "verkkosivu": null
      },
      "kuvaus": {
        "fi": "<p>Kuvaus</p>"
      }
    }
  ],
  "metadata": {
    "jarjestaaUrheilijanAmmKoulutusta": false,
    "valintakokeidenYleiskuvaus": {
      "fi": "<p>Valintakokeiden kuvaus - fi</p>"
    },
    "valintaperusteenValintakokeidenLisatilaisuudet": [
      {
        "id": "testi-id-123",
        "tilaisuudet": [
          {
            "osoite": {
              "osoite": {
                "fi": "osoite"
              },
              "postinumeroKoodiUri": "posti_00350#2"
            },
            "aika": {
              "alkaa": "2019-04-02T10:45",
              "paattyy": "2019-04-02T19:00"
            },
            "lisatietoja": {
              "fi": "<p>lisatietoja</p>"
            },
            "jarjestamispaikka": {
              "fi": "paikka"
            }
          }
        ]
      }
    ],
    "kynnysehto": {},
    "aloituspaikat": {
      "lukumaara": 150,
      "ensikertalaisille": 49,
      "kuvaus": {}
    },
    "kaytetaanHaunAlkamiskautta": false,
    "koulutuksenAlkamiskausi": {
      "alkamiskausityyppi": "tarkka alkamisajankohta",
      "koulutuksenAlkamispaivamaara": "2021-04-16T00:00",
      "koulutuksenPaattymispaivamaara": "2021-12-12T00:00",
      "koulutuksenAlkamiskausiKoodiUri": null,
      "koulutuksenAlkamisvuosi": null,
      "henkilokohtaisenSuunnitelmanLisatiedot": {}
    },
    "hakukohteenLinja": null
  },
  "valintakokeet": [
    {
      "tyyppiKoodiUri": "valintakokeentyyppi_1#1",
      "nimi": {
        "fi": "Kokeen nimi - fi"
      },
      "metadata": {
        "tietoja": {
          "fi": "<p>Tietoa kokeesta - fi</p>"
        },
        "vahimmaispisteet": 10.03,
        "liittyyEnnakkovalmistautumista": true,
        "ohjeetEnnakkovalmistautumiseen": {
          "fi": "<p>Ohjeet ennakkovalmistautumiseen - fi</p>"
        },
        "erityisjarjestelytMahdollisia": true,
        "ohjeetErityisjarjestelyihin": {
          "fi": "<p>Ohjeet erityisj√§rjestelyihin - fi</p>"
        }
      },
      "tilaisuudet": [
        {
          "osoite": {
            "osoite": {
              "fi": "fi osoite"
            },
            "postinumeroKoodiUri": "posti_00350#2"
          },
          "aika": {
            "alkaa": "2019-04-16T08:44",
            "paattyy": "2019-04-18T08:44"
          },
          "lisatietoja": {
            "fi": "<p>fi lisatietoja</p>"
          },
          "jarjestamispaikka": {
            "fi": "Jarjestamispaikka - fi"
          }
        }
      ]
    }
  ],
  "hakuajat": [
    {
      "alkaa": "2011-11-11T10:30",
      "paattyy": "2011-11-12T11:45"
    }
  ],
  "muokkaaja": "1.2.246.562.24.62301161440",
  "organisaatioOid": "1.2.246.562.10.52251087186",
  "kielivalinta": [
    "fi"
  ],
  "kaytetaanHaunHakulomaketta": false,
  "hakulomaketyyppi": "ataru",
  "hakulomakeAtaruId": "12345",
  "hakulomakeKuvaus": {},
  "hakulomakeLinkki": {},
  "valintaperusteId": "649adb37-cd4d-4846-91a9-84b58b90f928",
  "externalId": null,
  "esikatselu": false,
  "jarjestyspaikkaOid": "1.2.246.562.10.45854578546",
  "liitteidenToimitusaika": null,
  "hakukohdeKoodiUri": null
}