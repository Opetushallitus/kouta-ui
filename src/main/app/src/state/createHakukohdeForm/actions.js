import { getFormValues, startSubmit, stopSubmit } from 'redux-form';
import get from 'lodash/get';

import { JULKAISUTILA } from '../../constants';
import { createSavingErrorToast, createSavingSuccessToast } from '../toaster';
import { getHakukohdeByValues, validate } from './utils';
import { isNonEmptyObject } from '../../utils';

const formName = 'createHakukohdeForm';
const getHakukohdeFormValues = getFormValues(formName);

const getOidsFromPathname = pathname => {
  const split = pathname.split('/').filter(p => !!p);

  return {
    organisaatioOid: split[1],
    toteutusOid: split[3],
    hakuOid: split[5],
  };
};

export const saveHakukohde = hakukohde => (
  dispatch,
  getState,
  { apiUrls, httpClient },
) => {
  return httpClient.put(apiUrls.url('kouta-backend.hakukohde'), hakukohde);
};

export const submit = ({ tila = JULKAISUTILA.TALLENNETTU } = {}) => async (
  dispatch,
  getState,
  { history },
) => {
  const state = getState();
  const values = getHakukohdeFormValues(state);
  const errors = validate({ values, tila });

  dispatch(startSubmit(formName));

  if (isNonEmptyObject(errors)) {
    dispatch(stopSubmit(formName, errors));
    dispatch(createSavingErrorToast());
    return;
  }

  const {
    me: { oid: muokkaaja },
  } = state;

  const { organisaatioOid, toteutusOid, hakuOid } = getOidsFromPathname(
    history.location.pathname,
  );

  const formData = getHakukohdeByValues(values);

  const hakukohde = {
    ...formData,
    organisaatioOid,
    toteutusOid,
    hakuOid,
    tila,
    muokkaaja,
  };

  let hakukohdeData;

  try {
    const { data } = await dispatch(saveHakukohde(hakukohde));

    hakukohdeData = data;
  } catch (e) {
    dispatch(stopSubmit(formName));
    dispatch(createSavingErrorToast());
    return;
  }

  dispatch(stopSubmit(formName));
  dispatch(createSavingSuccessToast());

  if (get(hakukohdeData, 'oid')) {
    history.push(`/hakukohde/${hakukohdeData.oid}/muokkaus`);
  } else {
    history.push('/');
  }
};
