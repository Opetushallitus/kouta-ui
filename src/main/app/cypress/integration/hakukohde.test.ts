import { createHakukohdeForm } from './hakukohdeForm/createHakukohdeForm';
import { editHakukohdeForm } from './hakukohdeForm/editHakukohde';

describe('createHakukohde', createHakukohdeForm);
describe('editHakukohde', editHakukohdeForm);
