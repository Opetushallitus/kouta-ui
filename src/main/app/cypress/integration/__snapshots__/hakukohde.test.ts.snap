exports[`createHakukohde > should be able to create ammatillinen hakukohde #0`] =
{
  "esikatselu": false,
  "externalId": null,
  "hakuOid": "4.1.1.1.1.1",
  "hakuajat": [
    {
      "alkaa": "2019-04-02T10:45",
      "paattyy": "2019-11-25T23:59"
    }
  ],
  "hakulomakeAtaruId": "lomake_1",
  "hakulomakeKuvaus": {},
  "hakulomakeLinkki": {},
  "hakulomaketyyppi": "ataru",
  "jarjestyspaikkaOid": "1.2.246.562.10.45854578546",
  "kaytetaanHaunAikataulua": false,
  "kaytetaanHaunHakulomaketta": false,
  "kielivalinta": [
    "fi"
  ],
  "liitteet": [
    {
      "kuvaus": {
        "fi": "<p>Kuvaus</p>"
      },
      "nimi": {
        "fi": "Nimi"
      },
      "toimitusaika": "2019-11-25T23:59",
      "toimitusosoite": {
        "osoite": {
          "osoite": {
            "fi": "Osoite"
          },
          "postinumeroKoodiUri": "posti_00350#2"
        },
        "sahkoposti": "sahkoposti@email.com"
      },
      "toimitustapa": "osoite",
      "tyyppiKoodiUri": "liitetyypitamm_4#1"
    }
  ],
  "liitteetOnkoSamaToimitusaika": false,
  "liitteetOnkoSamaToimitusosoite": false,
  "liitteidenToimitusaika": null,
  "liitteidenToimitusosoite": null,
  "liitteidenToimitustapa": null,
  "metadata": {
    "aloituspaikat": {
      "ensikertalaisille": null,
      "kuvaus": {},
      "lukumaara": 10
    },
    "hakukohteenLinja": null,
    "kaytetaanHaunAlkamiskautta": false,
    "koulutuksenAlkamiskausi": {
      "alkamiskausityyppi": "henkilokohtainen suunnitelma",
      "henkilokohtaisenSuunnitelmanLisatiedot": {
        "fi": "<p>Henkilökohtaisen suunnitelman lisätiedot</p>"
      },
      "koulutuksenAlkamiskausiKoodiUri": null,
      "koulutuksenAlkamispaivamaara": null,
      "koulutuksenAlkamisvuosi": null,
      "koulutuksenPaattymispaivamaara": null
    },
    "kynnysehto": {
      "fi": "<p>Kynnysehto</p>"
    },
    "valintakokeidenYleiskuvaus": {
      "fi": "<p>Valintakokeiden kuvaus</p>"
    },
    "valintaperusteenValintakokeidenLisatilaisuudet": [
      {
        "id": "testi-id-123",
        "tilaisuudet": [
          {
            "aika": {
              "alkaa": "2019-04-02T10:45",
              "paattyy": "2019-04-02T19:00"
            },
            "jarjestamispaikka": {
              "fi": "paikka"
            },
            "lisatietoja": {
              "fi": "<p>lisatietoja</p>"
            },
            "osoite": {
              "osoite": {
                "fi": "osoite"
              },
              "postinumeroKoodiUri": "posti_00350#2"
            }
          }
        ]
      }
    ]
  },
  "muokkaaja": "1.2.246.562.24.62301161440",
  "nimi": {
    "fi": "Hakukohteen nimi"
  },
  "organisaatioOid": "1.2.246.562.10.52251087186",
  "pohjakoulutusvaatimusKoodiUrit": [
    "pohjakoulutusvaatimuskouta_pk#1"
  ],
  "pohjakoulutusvaatimusTarkenne": {
    "fi": "<p>Tarkenne</p>"
  },
  "tila": "julkaistu",
  "toinenAsteOnkoKaksoistutkinto": true,
  "toteutusOid": "2.1.1.1.1.1",
  "valintakokeet": [
    {
      "metadata": {
        "erityisjarjestelytMahdollisia": true,
        "liittyyEnnakkovalmistautumista": true,
        "ohjeetEnnakkovalmistautumiseen": {
          "fi": "<p>ohjeet ennakkovalmistautumiseen</p>"
        },
        "ohjeetErityisjarjestelyihin": {
          "fi": "<p>ohjeet erityisjärjestelyihin</p>"
        },
        "tietoja": {
          "fi": "<p>Tietoa hakijalle</p>"
        },
        "vahimmaispisteet": 10.03
      },
      "nimi": {
        "fi": "nimi"
      },
      "tilaisuudet": [
        {
          "aika": {
            "alkaa": "2019-04-02T10:45",
            "paattyy": "2019-04-02T19:00"
          },
          "jarjestamispaikka": {
            "fi": "paikka"
          },
          "lisatietoja": {
            "fi": "<p>lisatietoja</p>"
          },
          "osoite": {
            "osoite": {
              "fi": "osoite"
            },
            "postinumeroKoodiUri": "posti_00350#2"
          }
        }
      ],
      "tyyppiKoodiUri": "valintakokeentyyppi_8#1"
    }
  ],
  "valintaperusteId": "649adb37-cd4d-4846-91a9-84b58b90f928"
};

exports[`createHakukohde > should be able to create korkeakoulu hakukohde #0`] =
{
  "esikatselu": false,
  "externalId": null,
  "hakuOid": "4.1.1.1.1.1",
  "hakuajat": [
    {
      "alkaa": "2019-04-02T10:45",
      "paattyy": "2019-11-25T23:59"
    }
  ],
  "hakulomakeAtaruId": "lomake_1",
  "hakulomakeKuvaus": {},
  "hakulomakeLinkki": {},
  "hakulomaketyyppi": "ataru",
  "jarjestyspaikkaOid": "1.2.246.562.10.45854578546",
  "kaytetaanHaunAikataulua": false,
  "kaytetaanHaunHakulomaketta": false,
  "kielivalinta": [
    "fi"
  ],
  "liitteet": [
    {
      "kuvaus": {
        "fi": "<p>Kuvaus</p>"
      },
      "nimi": {
        "fi": "Nimi"
      },
      "toimitusaika": "2019-11-25T23:59",
      "toimitusosoite": {
        "osoite": {
          "osoite": {
            "fi": "Osoite"
          },
          "postinumeroKoodiUri": "posti_00350#2"
        },
        "sahkoposti": "sahkoposti@email.com"
      },
      "toimitustapa": "osoite",
      "tyyppiKoodiUri": "liitetyypitamm_4#1"
    }
  ],
  "liitteetOnkoSamaToimitusaika": false,
  "liitteetOnkoSamaToimitusosoite": false,
  "liitteidenToimitusaika": null,
  "liitteidenToimitusosoite": null,
  "liitteidenToimitustapa": null,
  "metadata": {
    "aloituspaikat": {
      "ensikertalaisille": 5,
      "kuvaus": {},
      "lukumaara": 10
    },
    "hakukohteenLinja": null,
    "kaytetaanHaunAlkamiskautta": false,
    "koulutuksenAlkamiskausi": {
      "alkamiskausityyppi": "henkilokohtainen suunnitelma",
      "henkilokohtaisenSuunnitelmanLisatiedot": {
        "fi": "<p>Henkilökohtaisen suunnitelman lisätiedot</p>"
      },
      "koulutuksenAlkamiskausiKoodiUri": null,
      "koulutuksenAlkamispaivamaara": null,
      "koulutuksenAlkamisvuosi": null,
      "koulutuksenPaattymispaivamaara": null
    },
    "kynnysehto": {
      "fi": "<p>Kynnysehto</p>"
    },
    "valintakokeidenYleiskuvaus": {
      "fi": "<p>Valintakokeiden kuvaus</p>"
    },
    "valintaperusteenValintakokeidenLisatilaisuudet": [
      {
        "id": "testi-id-123",
        "tilaisuudet": [
          {
            "aika": {
              "alkaa": "2019-04-02T10:45",
              "paattyy": "2019-04-02T19:00"
            },
            "jarjestamispaikka": {
              "fi": "paikka"
            },
            "lisatietoja": {
              "fi": "<p>lisatietoja</p>"
            },
            "osoite": {
              "osoite": {
                "fi": "osoite"
              },
              "postinumeroKoodiUri": "posti_00350#2"
            }
          }
        ]
      }
    ]
  },
  "muokkaaja": "1.2.246.562.24.62301161440",
  "nimi": {
    "fi": "Hakukohteen nimi"
  },
  "organisaatioOid": "1.2.246.562.10.52251087186",
  "pohjakoulutusvaatimusKoodiUrit": [
    "pohjakoulutusvaatimuskouta_pk#1"
  ],
  "pohjakoulutusvaatimusTarkenne": {
    "fi": "<p>Tarkenne</p>"
  },
  "tila": "julkaistu",
  "toinenAsteOnkoKaksoistutkinto": false,
  "toteutusOid": "2.1.1.1.1.1",
  "valintakokeet": [
    {
      "metadata": {
        "erityisjarjestelytMahdollisia": true,
        "liittyyEnnakkovalmistautumista": true,
        "ohjeetEnnakkovalmistautumiseen": {
          "fi": "<p>ohjeet ennakkovalmistautumiseen</p>"
        },
        "ohjeetErityisjarjestelyihin": {
          "fi": "<p>ohjeet erityisjärjestelyihin</p>"
        },
        "tietoja": {
          "fi": "<p>Tietoa hakijalle</p>"
        },
        "vahimmaispisteet": 10.03
      },
      "nimi": {
        "fi": "nimi"
      },
      "tilaisuudet": [
        {
          "aika": {
            "alkaa": "2019-04-02T10:45",
            "paattyy": "2019-04-02T19:00"
          },
          "jarjestamispaikka": {
            "fi": "paikka"
          },
          "lisatietoja": {
            "fi": "<p>lisatietoja</p>"
          },
          "osoite": {
            "osoite": {
              "fi": "osoite"
            },
            "postinumeroKoodiUri": "posti_00350#2"
          }
        }
      ],
      "tyyppiKoodiUri": "valintakokeentyyppi_8#1"
    }
  ],
  "valintaperusteId": "649adb37-cd4d-4846-91a9-84b58b90f928"
};

exports[`editHakukohde > should be able to edit hakukohde #0`] =
{
  "alkamiskausiKoodiUri": "kausi_k#1",
  "alkamisvuosi": "2024",
  "esikatselu": false,
  "externalId": null,
  "hakuOid": "4.1.1.1.1.1",
  "hakuajat": [
    {
      "alkaa": "2011-11-11T10:30",
      "paattyy": "2011-11-12T11:45"
    }
  ],
  "hakulomake": {},
  "hakulomakeAtaruId": "12345",
  "hakulomakeKuvaus": {},
  "hakulomakeLinkki": {},
  "hakulomaketyyppi": "ataru",
  "jarjestyspaikkaOid": "1.2.246.562.10.45854578546",
  "kaytetaanHaunAikataulua": false,
  "kaytetaanHaunHakulomaketta": false,
  "kielivalinta": [
    "fi"
  ],
  "liitteet": [
    {
      "kuvaus": {
        "fi": "<p>Kuvaus</p>"
      },
      "nimi": {
        "fi": "Nimi"
      },
      "toimitusaika": "2011-11-11T10:30",
      "toimitusosoite": {
        "osoite": {
          "osoite": {
            "fi": "Osoite"
          },
          "postinumeroKoodiUri": "posti_00350#2"
        },
        "sahkoposti": "sahkoposti@email.com"
      },
      "toimitustapa": "osoite",
      "tyyppiKoodiUri": "liitetyypitamm_4#1"
    }
  ],
  "liitteetOnkoSamaToimitusaika": false,
  "liitteetOnkoSamaToimitusosoite": false,
  "liitteidenToimitusaika": null,
  "liitteidenToimitusosoite": null,
  "liitteidenToimitustapa": null,
  "metadata": {
    "aloituspaikat": {
      "ensikertalaisille": 49,
      "kuvaus": {},
      "lukumaara": 150
    },
    "hakukohteenLinja": null,
    "kaytetaanHaunAlkamiskautta": false,
    "koulutuksenAlkamiskausi": {
      "alkamiskausityyppi": "tarkka alkamisajankohta",
      "henkilokohtaisenSuunnitelmanLisatiedot": {},
      "koulutuksenAlkamiskausiKoodiUri": null,
      "koulutuksenAlkamispaivamaara": "2021-04-16T00:00",
      "koulutuksenAlkamisvuosi": null,
      "koulutuksenPaattymispaivamaara": "2021-12-12T00:00"
    },
    "kynnysehto": {},
    "valintakokeidenYleiskuvaus": {
      "fi": "<p>Valintakokeiden kuvaus - fi</p>"
    },
    "valintaperusteenValintakokeidenLisatilaisuudet": [
      {
        "id": "testi-id-123",
        "tilaisuudet": [
          {
            "aika": {
              "alkaa": "2019-04-02T10:45",
              "paattyy": "2019-04-02T19:00"
            },
            "jarjestamispaikka": {
              "fi": "paikka"
            },
            "lisatietoja": {
              "fi": "<p>lisatietoja</p>"
            },
            "osoite": {
              "osoite": {
                "fi": "osoite"
              },
              "postinumeroKoodiUri": "posti_00350#2"
            }
          }
        ]
      }
    ]
  },
  "modified": "2019-04-04T08:28",
  "muokkaaja": "1.2.246.562.24.62301161440",
  "muuPohjakoulutusvaatimus": {},
  "nimi": {
    "fi": "Hakukohteen nimi"
  },
  "oid": "6.1.1.1.1.1",
  "organisaatioOid": "1.2.246.562.10.52251087186",
  "pohjakoulutusvaatimusKoodiUrit": [
    "pohjakoulutusvaatimuskouta_pk#1"
  ],
  "pohjakoulutusvaatimusTarkenne": {
    "fi": "<p>Fi tarkenne</p>"
  },
  "tila": "tallennettu",
  "toinenAsteOnkoKaksoistutkinto": true,
  "toteutusOid": "2.1.1.1.1.1",
  "valintakokeet": [
    {
      "metadata": {
        "erityisjarjestelytMahdollisia": true,
        "liittyyEnnakkovalmistautumista": true,
        "ohjeetEnnakkovalmistautumiseen": {
          "fi": "<p>Ohjeet ennakkovalmistautumiseen - fi</p>"
        },
        "ohjeetErityisjarjestelyihin": {
          "fi": "<p>Ohjeet erityisjärjestelyihin - fi</p>"
        },
        "tietoja": {
          "fi": "<p>Tietoa kokeesta - fi</p>"
        },
        "vahimmaispisteet": 10.03
      },
      "nimi": {
        "fi": "Kokeen nimi - fi"
      },
      "tilaisuudet": [
        {
          "aika": {
            "alkaa": "2019-04-16T08:44",
            "paattyy": "2019-04-18T08:44"
          },
          "jarjestamispaikka": {
            "fi": "Jarjestamispaikka - fi"
          },
          "lisatietoja": {
            "fi": "<p>fi lisatietoja</p>"
          },
          "osoite": {
            "osoite": {
              "fi": "fi osoite"
            },
            "postinumeroKoodiUri": "posti_00350#2"
          }
        }
      ],
      "tyyppiKoodiUri": "valintakokeentyyppi_1#1"
    }
  ],
  "valintaperusteId": "649adb37-cd4d-4846-91a9-84b58b90f928"
};

exports[`createHakukohde > should be able to create hakukohde with muu hakulomake #0`] =
{
  "esikatselu": false,
  "externalId": null,
  "hakuOid": "4.1.1.1.1.1",
  "hakuajat": [],
  "hakulomakeAtaruId": null,
  "hakulomakeKuvaus": {},
  "hakulomakeLinkki": {
    "fi": "http://example.com"
  },
  "hakulomaketyyppi": "muu",
  "kaytetaanHaunAikataulua": true,
  "kaytetaanHaunHakulomaketta": false,
  "kielivalinta": [
    "fi"
  ],
  "liitteet": [],
  "liitteetOnkoSamaToimitusaika": false,
  "liitteetOnkoSamaToimitusosoite": false,
  "liitteidenToimitusaika": null,
  "liitteidenToimitusosoite": null,
  "liitteidenToimitustapa": null,
  "metadata": {
    "aloituspaikat": {
      "ensikertalaisille": null,
      "kuvaus": {},
      "lukumaara": null
    },
    "hakukohteenLinja": null,
    "kaytetaanHaunAlkamiskautta": true,
    "koulutuksenAlkamiskausi": null,
    "kynnysehto": {},
    "valintakokeidenYleiskuvaus": {},
    "valintaperusteenValintakokeidenLisatilaisuudet": []
  },
  "muokkaaja": "1.2.246.562.24.62301161440",
  "nimi": {
    "fi": "Koulutuskeskus Salpaus, jatkuva haku, eläintenhoitaja"
  },
  "organisaatioOid": "1.2.246.562.10.52251087186",
  "pohjakoulutusvaatimusKoodiUrit": [
    "pohjakoulutusvaatimuskouta_pk#1"
  ],
  "pohjakoulutusvaatimusTarkenne": {
    "fi": "<p>Tarkenne</p>"
  },
  "tila": "tallennettu",
  "toinenAsteOnkoKaksoistutkinto": false,
  "toteutusOid": "2.1.1.1.1.1",
  "valintaperusteId": null
};

exports[`createHakukohde > should be able to create hakukohde with "ei sähköistä" hakulomake #0`] =
{
  "esikatselu": false,
  "externalId": null,
  "hakuOid": "4.1.1.1.1.1",
  "hakuajat": [],
  "hakulomakeAtaruId": null,
  "hakulomakeKuvaus": {
    "fi": "<p>hakulomake kuvaus</p>"
  },
  "hakulomakeLinkki": {},
  "hakulomaketyyppi": "ei sähköistä",
  "kaytetaanHaunAikataulua": true,
  "kaytetaanHaunHakulomaketta": false,
  "kielivalinta": [
    "fi"
  ],
  "liitteet": [],
  "liitteetOnkoSamaToimitusaika": false,
  "liitteetOnkoSamaToimitusosoite": false,
  "liitteidenToimitusaika": null,
  "liitteidenToimitusosoite": null,
  "liitteidenToimitustapa": null,
  "metadata": {
    "aloituspaikat": {
      "ensikertalaisille": null,
      "kuvaus": {},
      "lukumaara": null
    },
    "hakukohteenLinja": null,
    "kaytetaanHaunAlkamiskautta": true,
    "koulutuksenAlkamiskausi": null,
    "kynnysehto": {},
    "valintakokeidenYleiskuvaus": {},
    "valintaperusteenValintakokeidenLisatilaisuudet": []
  },
  "muokkaaja": "1.2.246.562.24.62301161440",
  "nimi": {
    "fi": "Koulutuskeskus Salpaus, jatkuva haku, eläintenhoitaja"
  },
  "organisaatioOid": "1.2.246.562.10.52251087186",
  "pohjakoulutusvaatimusKoodiUrit": [
    "pohjakoulutusvaatimuskouta_pk#1"
  ],
  "pohjakoulutusvaatimusTarkenne": {
    "fi": "<p>Tarkenne</p>"
  },
  "tila": "tallennettu",
  "toinenAsteOnkoKaksoistutkinto": false,
  "toteutusOid": "2.1.1.1.1.1",
  "valintaperusteId": null
};

exports[`createHakukohde > should be able to create hakukohde with lukiolinjat #0`] =
{
  "esikatselu": false,
  "externalId": null,
  "hakuOid": "4.1.1.1.1.1",
  "hakuajat": [],
  "hakulomakeAtaruId": null,
  "hakulomakeKuvaus": {},
  "hakulomakeLinkki": {},
  "hakulomaketyyppi": null,
  "kaytetaanHaunAikataulua": true,
  "kaytetaanHaunHakulomaketta": true,
  "kielivalinta": [
    "fi"
  ],
  "liitteet": [],
  "liitteetOnkoSamaToimitusaika": false,
  "liitteetOnkoSamaToimitusosoite": false,
  "liitteidenToimitusaika": null,
  "liitteidenToimitusosoite": null,
  "liitteidenToimitustapa": null,
  "metadata": {
    "aloituspaikat": {
      "ensikertalaisille": null,
      "kuvaus": {},
      "lukumaara": null
    },
    "hakukohteenLinja": {
      "alinHyvaksyttyKeskiarvo": 3.5,
      "linja": null,
      "lisatietoa": {
        "fi": "<p>Lisätietoa</p>"
      },
      "painotetutArvosanat": [
        {
          "painokerroin": "1,5",
          "painotettuOppiaine": {
            "label": "A1 englanti",
            "value": "kieli_en#1"
          }
        },
        {
          "painokerroin": "2",
          "painotettuOppiaine": {
            "label": "Musiikki",
            "value": "oppiaineetyleissivistava_mu#1"
          }
        }
      ]
    },
    "kaytetaanHaunAlkamiskautta": true,
    "koulutuksenAlkamiskausi": null,
    "kynnysehto": {},
    "valintakokeidenYleiskuvaus": {},
    "valintaperusteenValintakokeidenLisatilaisuudet": []
  },
  "muokkaaja": "1.2.246.562.24.62301161440",
  "nimi": {
    "fi": "hakukohdelomake.lukionYleislinja"
  },
  "organisaatioOid": "1.2.246.562.10.52251087186",
  "pohjakoulutusvaatimusKoodiUrit": [
    "pohjakoulutusvaatimuskouta_pk#1"
  ],
  "pohjakoulutusvaatimusTarkenne": {
    "fi": "<p>Tarkenne</p>"
  },
  "tila": "tallennettu",
  "toinenAsteOnkoKaksoistutkinto": false,
  "toteutusOid": "2.1.1.1.1.1",
  "valintaperusteId": null
};
